incubator/elasticsearch-exporter/templates/service.yaml

```diff
@@ -0,0 +1,19 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: {{ template "elasticsearch-exporter.fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+    app: {{ template "elasticsearch-exporter.name" . }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - name: http-service
+      port: {{ .Values.service.externalPort }}
+      targetPort: {{ .Values.service.internalPort }}
```

Comment:
Remove `service.internalPort`. It doesn't make sense to have it configurable. The container port is hard-coded anyways. Just reference the named port (`http`). You can the rename `service.externalPort` to just `service.port`.